machine_learning_hep/utilities_plot.py

```diff
@@ -106,6 +128,27 @@ def fill2dhist(df_, histo, nvar1, nvar2):
     fill_hist(histo, arr2)
     return histo
 
+def fill2dweighed(df_, histo, nvar1, nvar2, weight):
+    """
+    Fill a TH2 histogram with two variables from a dataframe.
+    """
+    #df_rd = df_[[nvar1, nvar2]]
+    #arr2 = df_rd.values
+    #fill_hist(histo, arr2)
+    for row in df_.itertuples():
+        histo.Fill(getattr(row, nvar1), getattr(row, nvar2),  getattr(row, weight))
+    return histo
+
+def fillweighed(df_, histo, nvar1, weight):
+    """
+    Fill a TH2 histogram with two variables from a dataframe.
+    """
+    #df_rd = df_[[nvar1, nvar2]]
+    #arr2 = df_rd.values
+    #fill_hist(histo, arr2)
+    for row in df_.itertuples():
+        histo.Fill(getattr(row, nvar1), getattr(row, weight))
```

Comment:
we should add some protection against a TH2 being passsed

Reply:
Not fixed